pkg/cloudprovider/providers/gce/gce_loadbalancer.go

```diff
@@ -88,11 +88,18 @@ func LoadBalancerSrcRanges() []string {
 
 // GetLoadBalancer is an implementation of LoadBalancer.GetLoadBalancer
 func (gce *GCECloud) GetLoadBalancer(clusterName string, service *v1.Service) (*v1.LoadBalancerStatus, bool, error) {
-	loadBalancerName := cloudprovider.GetLoadBalancerName(service)
+	// For backword compatibility we had to fetch the loadBalancerName this way or else a simple assignment for service.Status.LoadBalancer.Name would be enough.
+	var loadBalancerName string
+	if len(service.Status.LoadBalancer.Name) > 0 {
+		loadBalancerName = service.Status.LoadBalancer.Name
```

Comment:
.. If we have to deal with `Status` going away, then we *have* to store the LBName (reliably) in `Spec` (or annotations) instead, right?

Afaics, there's either no way to recover LBName robustly across a migration when we lose `Status` - or we *force* providers to store the name robustly in some underlying provider tags mechanism, in which case we may as well just remove all the code that tries to derive it using any other mechanism and avoid having diverging sources of truth.